invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LocationList/components/InternalLocationList/InternalLocationList.js

```diff
@@ -1,37 +1,61 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { internalLocation as internalLocationApi } from '../../../../../common/api';
+import {
+  internalLocation as internalLocationApi,
+  item as itemApi,
+} from '../../../../../common/api';
 import { openRecordEditor } from '../../../../../common/urls';
 import { Error, Loader, ResultsTable } from '../../../../../common/components';
 import { Button } from 'semantic-ui-react';
 import { NewButton } from '../../../components/buttons';
 import { formatter } from '../../../../../common/components/ResultsTable/formatters';
+import { DeleteRecordModal } from '../../../../backoffice/components';
 import omit from 'lodash/omit';
 
 export default class InternalLocationList extends Component {
   constructor(props) {
     super(props);
-    this.fetchInternalLocations = this.props.fetchInternalLocations;
+    this.fetchInternalLocations = props.fetchInternalLocations;
+    this.deleteInternalLocation = props.deleteInternalLocation;
   }
 
   componentDidMount() {
     this.fetchInternalLocations();
   }
 
-  prepareData(data) {
-    return data.hits.map(row => {
-      let serialized = formatter.internalLocation.toTable(row);
-      serialized['Actions'] = (
+  _handleOnRefClick(itemPid) {
+    openRecordEditor(itemApi.url, itemPid);
+  }
+
+  _rowActions(ilocPid) {
+    return (
+      <>
         <Button
+          icon={'edit'}
           size="small"
-          content={'Edit'}
-          onClick={() =>
-            openRecordEditor(internalLocationApi.url, row.internal_location_pid)
-          }
+          title={'Edit Record'}
+          onClick={() => openRecordEditor(internalLocationApi.url, ilocPid)}
+        />
+        <DeleteRecordModal
+          headerContent={`Are you sure you want to delete the Internal Location
+            record with ID ${ilocPid}?`}
+          deleteFunction={() => this.deleteInternalLocation(ilocPid)}
```

Comment:
Probably missed this in the previous review but it might be a good idea to keep the callback prop names consistent and use:
```suggestion
          onDelete={() => this.deleteInternalLocation(ilocPid)}
```

Reply:
I couldn't agree more.